src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,77 @@
+package duke;
+
+import duke.commands.AddCommand;
+import duke.commands.ByeCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.exceptions.InvalidCommandException;
+
+
+/**
+ * Class that initiates the Parser object. Reads command in string given by the user and returns
+ * the appropriate Command object.
+ */
+public class Parser {
+    private static final String BYE_COMMAND = "bye";
+    private static final String LIST_COMMAND = "list";
+    private static final String DONE_COMMAND = "done";
+    private static final String TODO_COMMAND = "todo";
+    private static final String DEADLINE_COMMAND = "deadline";
+    private static final String EVENT_COMMAND = "event";
+    private static final String DELETE_COMMAND = "delete";
+    private static final String FIND_COMMAND = "find";
+
+    /**
+     * Reads command in string given by the user
+     * Returns the appropriate Command object.
+     * Throws InvalidCommandException if the command is unknown.
+     *
+     * @param fullCommand String which contains the command from the user.
+     * @return Command object of the given command.
+     * @throws InvalidCommandException If command is not recognisable.
+     */
+    public static Command parse(String fullCommand) throws InvalidCommandException {
```

Comment:
Organisation is quite clear! Perhaps the invalid command case could be checked at the beginning so that the "happy path" is more clear ?